The test `dirsrvtests/tests/suites/retrocl/basic_test.py::test_retrocl_trimming_interval` has failed (probably some minor test issue), rerunning it with the debug enabled.

```
        assert inst.searchErrorsLog("trim_changelog: removed 9 change records")
        for idx in range(2,11):
            pattern="changenumber=%d,cn=changelog" % idx
>           assert inst.searchAccessLog(pattern)
E           AssertionError: assert False
E            +  where False = <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>>('changenumber=2,cn=changelog')
E            +    where <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>> = <lib389.DirSrv object at 0x7fa6e5dff6a0>.searchAccessLog
```

--- a/dirsrvtests/tests/suites/retrocl/basic_test.py
+++ b/dirsrvtests/tests/suites/retrocl/basic_test.py
@@ -329,6 +329,95 @@ def test_retrocl_trimming(topology_st):
     # Clean up
     inst.config.set('nsslapd-errorlog-level', '0')
 
+def test_retrocl_trimming_interval(topology_st, request):
+    """Test retrocl trimming interval works
+
+    :id: 261a951d-8fe3-4b94-b933-c21806f795e7
+    :setup: Standalone Instance
+    :steps:
+        1. Enable Retro changelog
+        2. Configure trimming (trimming records older than 20sec)
+        3. Do a dummy update and wait 5sec
+        4. Do a extra updates
+        5. wait for 10sec and restart (after restart the dummy update will soon be trimmed)
+        6. wait 15sec so that the extra updates are trimmed
+        7. stop the instance
+        8. check that dummy/extra updates have been trimmed
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+        8. Success
+    """
+
+    inst = topology_st.standalone
+    # Configure plugin
+    log.info('Configure retrocl plugin')
+    rcl = RetroChangelogPlugin(inst)
+    rcl.replace('nsslapd-include-suffix', DEFAULT_SUFFIX)
+    rcl.enable()
+    inst.restart()
+
+    # Do some updates
+    suffix = Domain(inst, DEFAULT_SUFFIX)
+
+    # Configure trimming
+    rcl.replace('nsslapd-changelog-trim-interval', '2')
+    rcl.replace('nsslapd-changelogmaxage', '20s')
+    inst.restart()
+
+    # this update will be the first one to be trimmed
+    # and will initialize in the last_trim time.
+    # This initialization is important to trigger the bug
+    # if not initialized it will trim immediately
+    suffix.replace('description', 'dummy')
+
+    # retrieve the index of the update in retroCL as
+    # starting point for the extra updates
+    try:
+        retro_changelog_suffix = DSLdapObjects(inst, basedn=RETROCL_SUFFIX)
+        cl_index = int(str(retro_changelog_suffix.filter(f'(targetDn={DEFAULT_SUFFIX})')[0].get_attr_vals_utf8('changeNumber')[0]))
+    except ldap.LDAPError as e:
+        log.fatal("Changelog search failed, error: " + str(e))
+        assert False
+    time.sleep(5)
+
+    # Those extra updates should be trimmed 5 sec after the previous one
+    for idx in range(0, 10):
+        suffix.replace('description', str(idx))
+
+    # wait for ~14s (sleep+stop+start), so the trimming
+    # of the first update occurs soon after the restart
+    time.sleep(10)
+    inst.config.set('nsslapd-plugin-logging', 'on')
+    inst.config.set('nsslapd-accesslog-level','260')
+    inst.stop()
+    inst.start()
+
+    # wait for an additional 15s.
+    # at that time we should have trim the extra updates
+    # checking both error logs and DEL operations in access logs
+    inst.config.set('nsslapd-errorlog-level', '65536') # plugin logging
+    time.sleep(15)
+    inst.stop()
+    assert inst.searchErrorsLog("trim_changelog: removed 9 change records")
+    for idx in range(cl_index+1,cl_index+10):
+        pattern="changenumber=%d,cn=changelog" % idx
+        assert inst.searchAccessLog(pattern)
+
+    # Clean up
+    inst.restart()
+
+    def fin():
+        inst.config.set('nsslapd-errorlog-level', '0')
+        inst.config.set('nsslapd-plugin-logging', 'off')
+        inst.config.set('nsslapd-accesslog-level','256')
+
+    request.addfinalizer(fin)
 
 if __name__ == '__main__':
     # Run isolated

--- a/ldap/servers/plugins/retrocl/retrocl_trim.c
+++ b/ldap/servers/plugins/retrocl/retrocl_trim.c
@@ -17,13 +17,13 @@
 typedef struct _trim_status
 {
     time_t ts_c_max_age;     /* Constraint  - max age of a changelog entry */
+    int ts_c_trim_interval;  /* Constraint  - interval to evaluate the need to trim */
     time_t ts_s_last_trim;   /* Status - last time we trimmed */
     int ts_s_initialized;    /* Status - non-zero if initialized */
     int ts_s_trimming;       /* non-zero if trimming in progress */
     PRLock *ts_s_trim_mutex; /* protects ts_s_trimming */
 } trim_status;
 static trim_status ts = {0L, 0L, 0, 0, NULL};
-static int trim_interval = DEFAULT_CHANGELOGDB_TRIM_INTERVAL; /* in second */
 
 /*
  * All standard changeLogEntry attributes (initialized in get_cleattrs)
@@ -239,12 +239,13 @@ trim_changelog(void)
     time_t now_maxage; /* used for checking if the changelog entry can be trimmed */
     changeNumber first_in_log = 0, last_in_log = 0;
     int num_deleted = 0;
-    int max_age, last_trim;
+    int max_age, last_trim, trim_interval;
 
     now_interval = slapi_current_rel_time_t(); /* monotonic time for interval */
 
     PR_Lock(ts.ts_s_trim_mutex);
     max_age = ts.ts_c_max_age;
+    trim_interval = ts.ts_c_trim_interval;
     last_trim = ts.ts_s_last_trim;
     PR_Unlock(ts.ts_s_trim_mutex);
 
@@ -366,7 +367,7 @@ retrocl_housekeeping(time_t cur_time, void *noarg __attribute__((unused)))
         int must_trim = 0;
         /* See if we need to trim */
         /* Has enough time elapsed since our last check? */
-        if (cur_time - ts.ts_s_last_trim >= (ts.ts_c_max_age)) {
+        if (cur_time - ts.ts_s_last_trim >= (ts.ts_c_trim_interval)) {
             /* Is the first entry too old? */
             time_t first_time;
             time_t now_maxage = slapi_current_utc_time(); /* real time for trimming candidates */
@@ -422,6 +423,7 @@ retrocl_init_trimming(void)
     const char *cl_maxage;
     time_t ageval = 0; /* Don't trim, by default */
     const char *cl_trim_interval;
+    int trim_interval;
 
     cl_maxage = retrocl_get_config_str(CONFIG_CHANGELOG_MAXAGE_ATTRIBUTE);
     if (cl_maxage) {
@@ -441,7 +443,7 @@ retrocl_init_trimming(void)
     cl_trim_interval = retrocl_get_config_str(CONFIG_CHANGELOG_TRIM_INTERVAL);
     if (cl_trim_interval) {
         trim_interval = strtol(cl_trim_interval, (char **)NULL, 10);
-        if (0 == trim_interval) {
+        if (0 >= trim_interval) {
             slapi_log_err(SLAPI_LOG_ERR, RETROCL_PLUGIN_NAME,
                           "retrocl_init_trimming: ignoring invalid %s value %s; "
                           "resetting the default %d\n",
@@ -453,6 +455,7 @@ retrocl_init_trimming(void)
     }
 
     ts.ts_c_max_age = ageval;
+    ts.ts_c_trim_interval = trim_interval;
     ts.ts_s_last_trim = (time_t)0L;
     ts.ts_s_trimming = 0;
     if ((ts.ts_s_trim_mutex = PR_NewLock()) == NULL) {
@@ -466,7 +469,7 @@ retrocl_init_trimming(void)
     retrocl_trim_ctx = slapi_eq_repeat_rel(retrocl_housekeeping,
                                            NULL, (time_t)0,
                                            /* in milliseconds */
-                                           trim_interval * 1000);
+                                           ts.ts_c_trim_interval * 1000);
 }
 
 /*

